.github/workflows/cd-analytics-infra.yml

```
@@ -0,0 +1,61 @@
+name: Deploy Analytics Infra
+# Todo: create a composite action
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "infra/analytics/**"
+
+jobs:
+  build-repository:
+    name: Build Repository
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.7.1
+          terraform_wrapper: false
+
+      - name: Configure AWS credentials
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          app_name: analytics
+          environment: shared
+      - name: Build Repository
+        run: |
+          ./bin/terraform-init-and-apply.sh infra/analytics/build-repository shared
+
+  deploy-infra:
+    name: Deploy Infrastructure
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        directory: ["database", "service"]
+        envs: [ "dev"]
```

```suggestion
        envs: ${{ (github.event_name == 'release' && ['prod']) || ['dev', 'staging'] }}
```

This needs to conditionally deploy to either `prod` or `['staging', and 'dev']` depending on whether or not this is a release.

See example: https://github.com/HHS/simpler-grants-gov/blob/d3de3f6f38153b7502844bfb17ef11b15b3a5c45/.github/workflows/cd-api.yml#L36C20-L36C100

This change needs to be in the other files as well